### PR TITLE
Made Navigation Drawer 

### PR DESCRIPTION
**Pull request by tim-macphail**
### Features
- Navigation drawer with working upload and export buttons
- Loading animation for files
- Server now sends a different format of json where the object keys are course names and the values are topic and assessments
```json
{
  "CPSC 413": {
    "topic": "unknown topic",
    "assessments": [...]
  },
  "SENG 550": ...
}
```

### Limitations
- No error handling on uploading wrong files
- Uploading two of the same file at once effectively ignores one of them
- Export button only does ics, no xslsx for now
- Gives random shapes from [circle, square, pentagon] for course icon. There does not appear to be a triangle easily available in material-icons

### Clips:
https://user-images.githubusercontent.com/70448914/217292165-86c376bb-36b0-4b5f-90c9-0636a600321d.mp4

https://user-images.githubusercontent.com/70448914/217292907-9f92056c-ba34-474a-adec-da00fe664670.mp4

### CODE REVIEW

File: client/src/components/NavigationDrawer/NavigationDrawer.tsx [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+              <div className="flex flex-col ml-2 min-w-0">
+                <p className={classnames("truncate")}>
+                  {file}
+                </p>
+              </div>
+            </Button>
+          ))}
+        </div>
+      )}
+      <Button
```
Comment (cloudyyoung) [inline client/src/components/NavigationDrawer/NavigationDrawer.tsx:105]: This button is implemented more complex that it should be, in terms of design/css.

File: client/src/logic/icsGen.ts [hunk 2]
```diff
@@ -12,22 +12,31 @@ export interface Course {
   assessments: Assessment[];
 }
 
-function jsonToICS(semester: Course[]): string {
+export interface CourseData {
+  topic: string;
+  assessments: Assessment[];
+}
+
+export interface Response {
```
Comment (cloudyyoung) [inline client/src/logic/icsGen.ts:20]: Please rename class names as they're quite confusing. And `Course` seems to not be used anywhere.

Consider following changes:
- `Course` -> `CourseInfo`
- `CourseData` -> `Course`, and the class contains `info: CourseInfo` and `assessments: Assessment[]`; **or** only `Course` which contains: `topic: string`, `description: string`, `hours: string`, etc and `assessments: Assessment[]`.
- `Response` -> `Courses`

If you wish to have a class `Response` for http requests, consider having a wrapping class like:
```
interface Response {
    statusCode: int;
    courses?: Courses;
    error?: string;
    ....
}
```

File: client/src/pages/Review/Review.tsx [hunk 3]
```diff
@@ -1,28 +1,26 @@
...
-  const courses = ["cpsc-413", "seng-513", "cpsc-481", "cpsc-449", "cpsc-457"];
   const [selectedTab, setSelectedTab] = useState(0);
-  const { courseId } = useParams();
-  const format = (courseId: string | undefined) => courseId?.replace("-", " ").toUpperCase();
+  const displayFormat = (course: string) => (course.replace(/-/g, " ").toUpperCase());
+
+  let { courseId } = useParams();
+  courseId = courseId ? displayFormat(courseId) : undefined;
+
+  const [data, setData] = useState<Response>({});
```
Comment (cloudyyoung) [inline client/src/pages/Review/Review.tsx:15]: Very bad variable name.

**Review by cloudyyoung — CHANGES_REQUESTED** [review]

File: client/src/components/NavigationDrawer/NavigationDrawer.tsx [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+  };
+
+  return (
+    <div className="flex flex-col w-full md:p-4 p-0 bg-surface">
+      <p className="m-5 ml-2 font-bold">Courses</p>
+
+      {data && Object.entries(data).map(([course, courseData]) => (
+        <Button
+          variant="text"
+          to={`/review/${urlFormat(course)}`}
```
[the remark this answers is not in the document]
Comment (tim-macphail) [reply client/src/components/NavigationDrawer/NavigationDrawer.tsx:55]: With this, if I click CPSC 413 from CPSC 331 i will end up at `/cpsc-331/cpsc-413` and get an empty page

File: client/src/components/NavigationDrawer/NavigationDrawer.tsx (hunk 1 above)
Comment (tim-macphail) [reply client/src/components/NavigationDrawer/NavigationDrawer.tsx:105]: disabled prop 👍

**Review by cloudyyoung — APPROVED** [review]

**Pull request merged**